components/scream/src/share/field/field_property_checks/field_nan_check.hpp

```diff
@@ -107,6 +107,27 @@ class FieldNaNCheck: public FieldPropertyCheck<RealType> {
           }, Kokkos::Sum<int>(num_nans));
         }
         break;
+      case 6:
+        {
+          auto v = field.template get_view<const_RT******>();
+          const int dim1 = dims[1];
+          const int dim2 = dims[2];
+          const int dim3 = dims[3];
+          const int dim4 = dims[4];
+          const int dim5 = dims[5];
+          Kokkos::parallel_reduce(dim0*dim1*dim2*dim3*dim4*dim5, KOKKOS_LAMBDA(int idx, int& result) {
+            const int i = ((((idx / dim5) / dim4) / dim3) / dim2) / dim1;
```

Comment:
Do you think it's worth having an `unflatten(index, rank, dims, indices)` method on the field or layout object with the following signature, in which `dims` and `indices` are of length `rank`, and `i`, `j`, `k`, etc are stored in `indices` as a result of the method?

```
void unflatten(int flattened_index, int rank, int dims[], int indices[])
```

Reply:
Uhm, it might be useful, but I'm worried about how to use it from device. I am not 100% sure, but I don't think device lambdas can capture host static arrays (such as dims). Though we could declare `int indices[6];` locally, and call `unflatten(idx, 6, dim0, dim1,dim2,dim3,dim4,dim5,indices);`.

If the kokkos layout stored a 1d view for the (physical) extents, then we could probably do it more easily:
```
const auto& layout = f.get_header().get_layout();
const auto extents = layout.extents(); // or dims_view(), or whatever
Kokkos::parallel_for(layout.size(),KOKKOS_LAMBDA(const int idx){
  int indices[6];
  unflatten_indices<6> (idx, extents, indices);
  ...
});
```
Imho, it's a good v1.1 issue.

Reply:
We could offer also an overload where indices are N separate ints, for ease of use. Like:

```
int indices[6];
int i,j,k,l,m,n;
unflatten<6>(idx,extents,indices); // with array
unflatten<6>(idx,extents,i,j,k,l,m,n);
v(i,j,k,l,m,n) = blah; // Concise
v(indices[0], indices[1], indices[2], indices[3], indices[4], indices[5]) = blah; // Lost in parsing...
```